frontend/src/components/Projectile.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { useEffect, useState } from "react";
+
+interface Props {
+  source: string;
+  target: string;
+  duration: number;
+  positiveNews: boolean;
+}
+
+interface Coordinates {
+  x: number;
+  y: number;
+}
+
+const center = (element: HTMLElement | null): Coordinates => {
+  if (!element) return { x: 0, y: 0 };
+
+  const bb = element.getBoundingClientRect();
+  return {
+    x: bb.x + bb.width / 2,
+    y: bb.y + bb.height / 2,
+  };
+};
+
+const Projectile: React.FunctionComponent<Props> = ({
+  source,
+  target,
+  duration,
+  positiveNews,
+}: Props) => {
+  const sourceEl = document.getElementById(`board-unit-${source}`);
+  const targetEl = document.getElementById(`board-unit-${target}`);
+  const sourceCenter = center(sourceEl);
+  const targetCenter = center(targetEl);
+  const dx = targetCenter.x - sourceCenter.x;
+  const dy = targetCenter.y - sourceCenter.y;
+  const angle = -Math.atan2(dx, dy);
+
+  const [isStarted, setStarted] = useState<boolean>(false);
+
+  useEffect(() => {
+    console.log({ sourceCenter, targetCenter });
+    setStarted(true);
+  }, []);
+
+  const transform = isStarted
```

Comment:
Could you explain what is this for?

Reply:
Basically we have to set the transform twice to make it animate. So on the initial render, `isStarted` is `false` so we use the source transform. Immediately `useEffect` fires and updates `isStarted` to `true`, so we use the target transform. I chose `isStarted` as the name as it refers to the animation state,